nomad/structs/alloc.go

```diff
@@ -486,16 +486,6 @@ func (a *Allocation) NextRescheduleTimeByTime(t time.Time) (time.Time, bool) {
 	return a.nextRescheduleTime(t, reschedulePolicy)
 }
 
-func (a *Allocation) RescheduleTimeOnDisconnect(now time.Time) (time.Time, bool) {
-	tg := a.Job.LookupTaskGroup(a.TaskGroup)
-	if tg == nil || tg.Disconnect == nil || tg.Disconnect.Replace == nil {
-		// Kept to maintain backwards compatibility with behavior prior to 1.8.0
-		return a.NextRescheduleTimeByTime(now)
-	}
-
-	return now, *tg.Disconnect.Replace
-}
```

Comment:
We can get rid of this outright because we want to make sure we're respecting the reschedule delay, right?